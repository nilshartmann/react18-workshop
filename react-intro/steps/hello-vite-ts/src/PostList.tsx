import moment from "moment";
import { BlogPost } from "./types";

function formattedDate(date: string) {
  return moment(date).format("DD.MM.YYYY");
}

type PostListProps = {
  posts: BlogPost[];
  onAddPost(): void;
};

export default function PostList({ posts, onAddPost }: PostListProps) {
  return (
    <>
      <button onClick={onAddPost}>Add Post</button>
      {posts.map(p => (
        <article key={p.id} className="Container">
          <p className="Date">{formattedDate(p.date)}</p>
          <h1>{p.title}</h1>
          <p>{p.body}</p>
          {p.tags?.length > 0 && (
            <div className="Tags">
              <b>Tags: </b>
              {p.tags?.join(", ")}
            </div>
          )}
        </article>
      ))}
    </>
  );
}
