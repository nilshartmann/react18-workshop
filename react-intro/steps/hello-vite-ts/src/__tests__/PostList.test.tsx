import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import PostList from "../PostList";
import { expect, it } from "vitest";

const mockPosts = [
  {
    id: "1",
    title: "One",
    body: "Lorem ipsum",
    date: "2019-08-23T18:25:43.511Z",
    tags: ["Dummy", "Content"]
  },
  {
    id: "2",
    title: "Second Post",
    body: "Some more content",
    date: "2019-09-22T14:12:21.511Z",
    tags: []
  }
];

it("renders correctly", () => {
  const result = render(<PostList posts={mockPosts} />);
  expect(result.asFragment()).toMatchSnapshot();
});
