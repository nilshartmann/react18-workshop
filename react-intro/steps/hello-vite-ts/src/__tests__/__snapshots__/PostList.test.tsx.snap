// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div>
    <article
      class="Container"
    >
      <p
        class="Date"
      >
        08/23/2019
      </p>
      <h1>
        One
      </h1>
      <p>
        Lorem ipsum
      </p>
    </article>
    <article
      class="Container"
    >
      <p
        class="Date"
      >
        09/22/2019
      </p>
      <h1>
        Second Post
      </h1>
      <p>
        Some more content
      </p>
    </article>
  </div>
</DocumentFragment>
`;
